### PR TITLE
Added Site Reporting Rates Report for pnlppgi domain

### PR DESCRIPTION
**Pull request by lwyszomi**
@NoahCarnahan

### CODE REVIEW

File: custom/pnlppgi/expressions.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    type = TypeProperty('year_expression')
+    property = StringProperty()
+
+    def __call__(self, item, context=None):
+        date = force_to_datetime(item[self.property])
+        return date.year
+
+
+def week_expression(spec, context):
+    wrapped = w=WeekExpressionSpec.wrap(spec)
```
Comment (dimagimon) [inline custom/pnlppgi/expressions.py]: F841 local variable 'w' is assigned to but never used
E225 missing whitespace around operator

File: custom/pnlppgi/expressions.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    return wrapped
+
+
+def year_expression(spec, context):
+    wrapped = YearExpressionSpec.wrap(spec)
+    return wrapped
```
Comment (dimagimon) [inline custom/pnlppgi/expressions.py]: W292 no newline at end of file

File: custom/pnlppgi/reports.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
+import datetime
+from sqlagg.columns import SimpleColumn, CountColumn
+from sqlagg.filters import LT, EQ
```
Comment (dimagimon) [inline custom/pnlppgi/reports.py]: F401 'LT' imported but unused

File: settings.py [hunk 4]
```diff
@@ -893,7 +894,7 @@
 ZIPLINE_API_USER = ''
 ZIPLINE_API_PASSWORD = ''
 
-KAFKA_URL = 'localhost:9092'
+KAFKA_URL = 'localhost:1111'
```
Comment (NoahCarnahan) [inline settings.py]: Looks like this line might have been committed inadvertently

**Comment by NoahCarnahan** [comment]
Just one comment, but otherwise looks good. Could you also add the static ucr datasource configuration for the report to this PR?

**Comment by lwyszomi** [comment]
@NoahCarnahan 
I wanted to add a data source by UI, but If you want static I will add.

File: settings.py (hunk 4 above)
Comment (lwyszomi) [inline settings.py]: I will remove this change.

**Comment by NoahCarnahan** [comment]
I think a static data source would be preferable in case some one else accidentally deletes it from the domain, or something like that. Thanks!

**Comment by lwyszomi** [comment]
@NoahCarnahan updated

File: custom/pnlppgi/reports.py [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+
+    @property
+    def group_by(self):
+        return ['site_id']
+
+    @property
+    def columns(self):
+
+        def percent(x):
+            return {'sort_key': x, 'html': '%.2f%%' % (x*100)}
```
Comment (dimagimon) [inline custom/pnlppgi/reports.py]: E226 missing whitespace around arithmetic operator

File: custom/pnlppgi/reports.py [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+            DatabaseColumn('cas_suspects_fe', SumColumn('cas_suspects_fe')),
+            DatabaseColumn('tests_realises_fe', SumColumn('tests_realises_fe')),
+            DatabaseColumn('cas_confirmes_fe', SumColumn('cas_confirmes_fe')),
+            DatabaseColumn('cas_vu_total', SumColumn('cas_vu_total')),
+            DatabaseColumn('cas_suspect_total', SumColumn('cas_suspect_total')),
+            DatabaseColumn('tests_realises_total', SumColumn('tests_realises_total')),
+            DatabaseColumn('cas_confirmes_total', SumColumn('cas_confirmes_total')),
+            AggregateColumn(
+                'div_teasts_cas',
+                lambda x, y: (x or 0)/float(y or 1),
```
Comment (dimagimon) [inline custom/pnlppgi/reports.py]: E226 missing whitespace around arithmetic operator

File: custom/pnlppgi/reports.py [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+                        row.get('cas_suspects_fe', EMPTY_CELL),
+                        row.get('tests_realises_fe', EMPTY_CELL),
+                        row.get('cas_confirmes_fe', EMPTY_CELL),
+                        row.get('cas_vu_total', EMPTY_CELL),
+                        row.get('cas_suspect_total', EMPTY_CELL),
+                        row.get('tests_realises_total', EMPTY_CELL),
+                        row.get('cas_confirmes_total', EMPTY_CELL),
+                        row.get('div_teasts_cas', EMPTY_CELL)
+                    ]
+
```
Comment (dimagimon) [inline custom/pnlppgi/reports.py]: W391 blank line at end of file

**Comment by lwyszomi** [comment]
@NoahCarnahan I added weekly malaria report.

**Comment by lwyszomi** [comment]
@NoahCarnahan Also I added cumulative report.

**Comment by lwyszomi** [comment]
@NoahCarnahan pls don't merge I have some problems with chart.

File: custom/pnlppgi/reports.py [hunk 8]
```diff
@@ -0,0 +1,466 @@
...
+class SiteReportingRatesReport(SqlTabularReport, CustomProjectReport, ProjectReportParametersMixin):
+    slug = 'site_reporting_rates_report'
+    name = 'Site Reporting Rates Report'
+
+    report_template_path = 'pnlppgi/site_reporting.html'
+
+    @use_nvd3
+    def decorator_dispatcher(self, request, *args, **kwargs):
+        return super(SiteReportingRatesReport, self).decorator_dispatcher(request, *args, **kwargs)
+
```
Comment (dimagimon) [inline custom/pnlppgi/reports.py]: W293 blank line contains whitespace

**Comment by lwyszomi** [comment]
@NoahCarnahan ready to review and merge. I will add ZONES in Cumulative Malaria report when hierarchy will be updated on production.

**Comment by lwyszomi** [comment]
screen shot 2016-09-15 at 12 49 16
screen shot 2016-09-15 at 12 49 56
screen shot 2016-09-15 at 14 59 34

**Comment by NoahCarnahan** [comment]
Looks great, thanks!

**Comment by czue** [comment]
very nice

**Pull request merged**